controllers/machinehealthcheck_targets_test.go

```diff
@@ -327,7 +332,274 @@ func TestHealthCheckTargets(t *testing.T) {
 	}
 }
 
+func TestRemediate(t *testing.T) {
+	namespace := "test-mhc"
+	clusterName := "test-cluster"
+	labels := map[string]string{"cluster": clusterName, "machine-group": "foo"}
+
+	machineSetORs := []metav1.OwnerReference{{Kind: "MachineSet"}}
+
+	workerNode := newTestNode("worker-node")
+	workerMachine := newTestMachine("worker-machine", namespace, clusterName, workerNode.Name, labels)
+	workerMachine.SetOwnerReferences(machineSetORs)
+	workerMachineUnowned := newTestMachine("worker-machine", namespace, clusterName, workerNode.Name, labels)
+
+	masterNode := newTestNode("master-node")
+	if masterNode.Labels == nil {
+		masterNode.Labels = make(map[string]string)
+	}
+	masterNode.Labels[nodeMasterLabel] = ""
+
+	masterMachine := newTestMachine("master-machine", namespace, clusterName, masterNode.Name, labels)
+	masterMachine.SetOwnerReferences(machineSetORs)
```

Comment:
A control plane machine should never be owned by a MachineSet. Having it owned by a KubeadmControlPlane would be a more accurate example.

Reply:
I've updated this. One point I note now is, the check for a machineset owner comes before the check for a control plane machine (which I think is correct). It made me realise however the control plane check is a little redundant, it will only ever protect against, if it ever happened, a control plane machine being owned by a machineset, which is a very non-standard way of running.

Perhaps the control plane check should be dropped? Though I can also see argument for leaving it in in-case we change the MachineSet requirement in the future, WDYT? (@enxebre do you have any opinions on this?)

Reply:
I think I agree with this. We could start by only remediating Machines that belong to a MachineSet (that would be the only check). And pending the decision on KubeadmControlPlane, we could expand it to only remediating Machines that have a controller owner (this would cover MachineSets and presumably all control plane implementations).

Reply:
I agree. I think it make sense to address KubeadmControlPlane separately and discuss all sensitive possible scenarios for control plane.

Reply:
@ncdc How would you like me to proceed on this comment, should I get rid of the `isControlPlane` checks and just rely on `MachineSet` ownership for now?

Reply:
I think it's fine to keep the check that excludes control plane machines until we're ready for control plane/etcd remediation.